crawl-ref/source/mon-ench.cc

```diff
@@ -317,14 +304,42 @@ void monster::add_enchantment_effect(const mon_enchant &ench, bool quiet)
         break;
 
     case ENCH_RING_OF_THUNDER:
-        _place_thunder_ring(*this);
-        if (you.see_cell(pos()))
+        surround_mons_with_cloud(this, CLOUD_STORM);
+        if (you.see_cell(pos()) && !quiet)
         {
             mprf(MSGCH_WARN, "A violent storm begins to rage around %s.",
                 name(DESC_THE).c_str());
         }
         break;
 
+    case ENCH_RING_OF_FLAMES:
+        surround_mons_with_cloud(this, CLOUD_FIRE);
+        break;
+
+    case ENCH_RING_OF_CHAOS:
+        surround_mons_with_cloud(this, CLOUD_CHAOS);
+        break;
+
+    case ENCH_RING_OF_MUTATION:
+        surround_mons_with_cloud(this, CLOUD_MUTAGENIC);
+        break;
+
+    case ENCH_RING_OF_FOG:
+        surround_mons_with_cloud(this, CLOUD_GREY_SMOKE);
+        break;
+
+    case ENCH_RING_OF_ICE:
+        surround_mons_with_cloud(this, CLOUD_COLD);
+        break;
+
+    case ENCH_RING_OF_DRAINING:
+        surround_mons_with_cloud(this, CLOUD_NEGATIVE_ENERGY);
+        break;
+
+    case ENCH_RING_OF_ACID:
+        surround_mons_with_cloud(this, CLOUD_ACID);
+        break;
+
```

Comment:
These should be deduplicated, have a block of fallthroughs for each of these, then

```
// .... cases
case ENCH_RING_OF_ACID:
    update_mons_cloud_ring( /* args */);
    break;
```